src/quo/components/inputs/address_input/view.cljs

```diff
@@ -55,124 +54,135 @@
+                                          ens?     (when ens-regex
+                                                     (boolean (re-matches ens-regex text)))]
+                                      (reset! value text)
+                                      (if (> (count text) 0)
+                                        (set-status :typing)
+                                        (set-status :active))
+                                      (when on-change-text
+                                        (on-change-text text))
+                                      (when (and on-detect-ens ens?)
+                                        (set-status :loading)
+                                        (on-detect-ens text #(set-status :typing)))
+                                      (when (and address? on-detect-address)
+                                        (set-status :loading)
+                                        (on-detect-address text))
+                                      (when (and (not address?)
+                                                 (not ens?)
+                                                 on-detect-unclassified)
+                                        (on-detect-unclassified text)))))
+        set-value                (rn/use-callback
+                                  (fn [new-value]
+                                    (reset! value new-value)
+                                    (on-change new-value)
+                                    (trigger-render-value new-value)))
+        on-paste                 (rn/use-callback
+                                  (fn []
+                                    (clipboard/get-string
+                                     (fn [clipboard]
+                                       (when-not (empty? clipboard)
+                                         (set-value clipboard))))))
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-scan                  (when on-scan (rn/use-callback #(on-scan set-value)))
```

Comment:
why are we passing the callback to `on-scan`?

Reply:
you mean set-value ? because it's a proper way to do it, we expect that on-scan will scan something and result will be set in the text input